src/components/Trade/hooks/useSwapper/swapperManager.ts

```diff
@@ -26,18 +27,15 @@ import { getWeb3InstanceByChainId } from 'lib/web3-instance'
 import type { FeatureFlags } from 'state/slices/preferencesSlice/preferencesSlice'
 
 // singleton - do not export me, use getSwapperManager
-let _swapperManager: SwapperManager | null = null
+let _swapperManager: Promise<SwapperManager> | undefined
+
 // singleton - do not export me
 // Used to short circuit calls to getSwapperManager if flags have not changed
 let previousFlags: string = ''
 
-export const getSwapperManager = async (flags: FeatureFlags): Promise<SwapperManager> => {
-  const flagsChanged = previousFlags !== JSON.stringify(flags)
-  if (_swapperManager && !flagsChanged) return _swapperManager
-  previousFlags = JSON.stringify(flags)
-
```

Comment:
not an issue causing real bugs, BUT we should never compare objects from non-stable stringify. Fixed further down the page using `'fast-json-stable-stringify'`